chore: update README (#514)
ovh/the-bastion: README.md

:lock: The Bastion
==================

## Overview

Bastions are a cluster of machines used as the unique entry point by operational teams (such as sysadmins, developers, database admins, ...) to securely connect to devices (servers, virtual machines, cloud instances, network equipment, ...), usually using `ssh`.

Bastions provide mechanisms for authentication, authorization, traceability and auditability for your whole infrastructure.

> Being between your users and your infrastructure, bastions add a layer of abstraction in-between so that your infrastructure doesn't need to know your operational team members individually.

Reviewer: ```suggestion
Being between your users and your infrastructure, The Bastion adds a layer of abstraction in-between so that your infrastructure doesn't need to know your operational team members individually.
```

Each of your team member has an individual account on The Bastion, and may be a member of one or several bastion groups that may give them access to one or more infrastructures. The infrastructure devices only need to know and trust the bastion group(s) they may be a part of.

> The Bastion fine-grained RBAC makes it possible to delegate some responsibilities to any account, group-scoped or bastion-wide, including to accounts that might be used by your automation to e.g. managing the lifecycle of the accounts (linked to your human resources management system, your LDAP or AD), ensure a group's ACL is up to date (linked to your CMDB), etc. Automated processes are easy to implement through the [JSON API over SSH](https://ovh.github.io/the-bastion/using/api.html).

Reviewer: ```suggestion
The Bastion fine-grained RBAC makes it possible to delegate some responsibilities to any account, group-scoped or bastion-wide, including to accounts that might be used by your automation to e.g. manage the lifecycle of the accounts (linked to your human resources management system, your LDAP or AD), ensure a group's ACL is up to date (linked to your CMDB), etc. Automated processes are easy to implement through the [JSON API over SSH](https://ovh.github.io/the-bastion/using/api.html).
```

## Knowledge resources

Want to know more while viewing some nice drawings? Here is a series of blog posts that dig more into the core functionalities and principles of The Bastion:
- [Part 1 - Genesis](https://blog.ovhcloud.com/the-ovhcloud-bastion-part-1/)
- [Part 2 - Delegation Dizziness](https://blog.ovhcloud.com/the-ovhcloud-ssh-bastion-part-2-delegation-dizziness/)
- [Part 3 - Security at the Core](https://blog.ovhcloud.com/the-bastion-part-3-security-at-the-core/)
- [Part 4 - A new era](https://blog.ovhcloud.com/the-bastion-part-4-a-new-era/)

Other resources that might be of interest:
- [Online documentation](https://ovh.github.io/the-bastion/)

> - (Video in French, slides in English) [The Bastion at the Very Tech Trip 2023](https://verytechtrip.ovhcloud.com/fr-ca/sessions), case study of managing an infrastructure with and without a bastion

Reviewer: ```suggestion
- (Video in French, slides in English) [The Bastion at the Very Tech Trip 2023](https://verytechtrip.ovhcloud.com/fr-ca/sessions), case study of managing an infrastructure with and without The Bastion
```

- (Podcast in French) [The Bastion at NoLimitSecu, 2021](https://www.nolimitsecu.fr/the-bastion), interview with questions & answers

## :recycle: Zero assumptions on your environment

Nothing fancy is needed either on the ingress or the egress side of The Bastion to make it work.

Only your good old `ssh` client is needed to connect through it, and on the other side, any standard `sshd` server will do the trick. This includes, for example, network devices on which you may not have the possibility to install any custom software.

Ancient devices that only support low-security cryptography algorithms or telnet can be hidden from the Internet by firewalling them and only allowing The Bastion, hereby avoiding a low-security trade-off by still allowing only high-security grade connections on the bastion ingress side.

## :curly_loop: Reliability

* Only a few well-known libraries are used, less third party code means a tinier attack surface

> * The bastion is engineered to be self-sufficient: no dependencies such as databases, other daemons, other machines, or third-party cloud services, neither for the authentication or authorization phase, statistically means less downtime

Reviewer: ```suggestion
* The Bastion is engineered to be self-sufficient: no dependencies such as databases, other daemons, other machines, or third-party cloud services, neither for the authentication or authorization phase, statistically means less downtime
```

* High availability can be setup so that multiple bastion instances form a cluster of several instances, with any instance usable at all times (active/active scheme)

> ## :godmode: Non-exhaustive features list

Reviewer: ```suggestion
## :godmode: Non-exhaustive feature list
```

- [Personal](https://ovh.github.io/the-bastion/using/basics/access_management.html#personal-accesses) and [group](https://ovh.github.io/the-bastion/using/basics/access_management.html#group-accesses) access schemes with group roles delegation to ensure teams autonomy without security trade-offs
- SSH protocol break between the ingress and egress connections
- Interactive session recording (in standard ttyrec files)
- Non-interactive session recording (stdout and stderr through ttyrec)
- Extensive [logging](https://ovh.github.io/the-bastion/administration/logs.html) support through syslog for easy SIEM consumption

> - Authentication features include support for [MFA/2FA](https://ovh.github.io/the-bastion/administration/mfa.html) (password, TOTP) in addition to publickey authentication

Reviewer: ```suggestion
- Authentication features include support for [MFA/2FA](https://ovh.github.io/the-bastion/administration/mfa.html) (password, TOTP) in addition to public key authentication
```
